File: frontend/src/api/autopilot.ts
import api from '@/api'

// 与后端 dto/response/agent.go:AutopilotTaskResponse 对齐
export interface AutopilotTask {
  id: number
  workspace_id: number
  project_id: number | null
  name: string
  description: string
  trigger_type: 'cron' | 'webhook' | 'manual'
  cron_expression: string
  trigger_url: string
  task_type: string
  agent_template_id: number | null
  agent_config_id: number | null
  input_data: Record<string, any> | null
  status: string
  last_run_at: string | null
  next_run_at: string | null
  config: Record<string, any> | null
  notification_config: Record<string, any> | null
  timeout_seconds: number
  retry_count: number
  enabled: boolean
  created_at: string
  updated_at: string
}

// 与后端 dto/response/agent.go:AutopilotExecutionResponse 对齐
export interface AutopilotExecution {
  id: number
  task_id: number
  status: 'pending' | 'running' | 'completed' | 'failed'
  trigger_type: string
  input_data: Record<string, any> | null
  output_data: Record<string, any> | null
  error_info: string
  logs: Record<string, any> | null
  started_at: string | null
  completed_at: string | null
  failed_at: string | null
  duration_ms: number
  retry_count: number
  created_at: string
}

// 与后端 dto/request/agent.go:AutopilotTaskCreate 对齐
export interface AutopilotCreateRequest {
  name: string
  description?: string
  trigger_type: 'cron' | 'webhook' | 'manual'
  cron_expression?: string
  task_type: string
  agent_template_id?: number | null
  agent_config_id?: number | null
  project_id?: number | null
  input_data?: Record<string, any>
  config?: Record<string, any>
  notification_config?: Record<string, any>
  timeout_seconds?: number
  retry_count?: number
  enabled?: boolean
}

// 与后端 dto/request/agent.go:AutopilotTaskUpdate 对齐
export interface AutopilotUpdateRequest {
  name?: string
  description?: string
  cron_expression?: string
  input_data?: Record<string, any>
  config?: Record<string, any>
  enabled?: boolean
}

export const autopilotApi = {
  async list(workspaceId: number): Promise<AutopilotTask[]> {
    const response = await api.get(`/workspaces/${workspaceId}/autopilot-tasks`)
    return response.data
  },

  async get(workspaceId: number, taskId: number): Promise<AutopilotTask> {
    const response = await api.get(`/workspaces/${workspaceId}/autopilot-tasks/${taskId}`)
    return response.data
  },

  async create(workspaceId: number, data: AutopilotCreateRequest): Promise<AutopilotTask> {
    const response = await api.post(`/workspaces/${workspaceId}/autopilot-tasks`, data)
    return response.data
  },

  async update(workspaceId: number, taskId: number, data: AutopilotUpdateRequest): Promise<AutopilotTask> {
    const response = await api.put(`/workspaces/${workspaceId}/autopilot-tasks/${taskId}`, data)
    return response.data
  },

  async delete(workspaceId: number, taskId: number): Promise<void> {
    await api.delete(`/workspaces/${workspaceId}/autopilot-tasks/${taskId}`)
  },

  async toggle(workspaceId: number, taskId: number): Promise<AutopilotTask> {
    const response = await api.post(`/workspaces/${workspaceId}/autopilot-tasks/${taskId}/toggle`)
    return response.data
  },

  async execute(workspaceId: number, taskId: number): Promise<AutopilotExecution> {
    const response = await api.post(`/workspaces/${workspaceId}/autopilot-tasks/${taskId}/execute`)
    return response.data
  },

  async listExecutions(workspaceId: number, taskId: number): Promise<AutopilotExecution[]> {
    const response = await api.get(`/workspaces/${workspaceId}/autopilot-tasks/${taskId}/executions`)
    return response.data
  },

  async getExecution(workspaceId: number, taskId: number, executionId: number): Promise<AutopilotExecution> {
    const response = await api.get(`/workspaces/${workspaceId}/autopilot-tasks/${taskId}/executions/${executionId}`)
    return response.data
  }
}

File: backend/internal/model/autopilot.go
package model

import (
	"encoding/json"
	"time"
)

// AutopilotTask represents a scheduled/automatic task
type AutopilotTask struct {
	BaseModel
	WorkspaceID    uint64         `gorm:"not null;index" json:"workspace_id"`
	ProjectID      *uint64        `gorm:"index" json:"project_id,omitempty"`
	Name           string         `gorm:"size:100;not null" json:"name"`
	Description    string         `gorm:"type:text" json:"description"`
	TriggerType    string         `gorm:"size:20;not null" json:"trigger_type"` // cron, webhook, manual
	CronExpression string         `gorm:"size:100" json:"cron_expression"`
	TriggerURL     string         `gorm:"size:255" json:"trigger_url"`
	TaskType       string         `gorm:"size:50;not null" json:"task_type"` // report, scan, sync, etc.
	AgentTemplateID *uint64       `gorm:"index" json:"agent_template_id,omitempty"`
	AgentConfigID   *uint64       `gorm:"index" json:"agent_config_id,omitempty"`
	InputData      json.RawMessage `gorm:"type:jsonb;default:'{}'" json:"input_data"`
	Status         string         `gorm:"size:20;not null;default:active" json:"status"`
	LastRunAt      *time.Time     `json:"last_run_at,omitempty"`
	NextRunAt      *time.Time     `json:"next_run_at,omitempty"`
	Config         json.RawMessage `gorm:"type:jsonb;default:'{}'" json:"config"`
	NotificationConfig json.RawMessage `gorm:"type:jsonb;default:'{}'" json:"notification_config"`
	TimeoutSeconds int            `gorm:"default:300" json:"timeout_seconds"`
	RetryCount     int            `gorm:"default:0" json:"retry_count"`
	Enabled        bool           `gorm:"default:true" json:"enabled"`
	CreatedAt      time.Time      `json:"created_at"`
	UpdatedAt      time.Time      `json:"updated_at"`
	DeletedAt      *time.Time     `gorm:"index" json:"deleted_at,omitempty"`
}

// AutopilotExecution represents an execution record of an autopilot task
type AutopilotExecution struct {
	BaseModel
	TaskID         uint64         `gorm:"not null;index" json:"task_id"`
	Status         string         `gorm:"size:20;not null;default:pending" json:"status"`
	TriggerType    string         `gorm:"size:20" json:"trigger_type"`
	InputData      json.RawMessage `gorm:"type:jsonb;default:'{}'" json:"input_data"`
	OutputData     json.RawMessage `gorm:"type:jsonb" json:"output_data"`
	ErrorInfo      string         `json:"error_info"`
	Logs           json.RawMessage `gorm:"type:jsonb;default:'[]'" json:"logs"`
	StartedAt      *time.Time     `json:"started_at,omitempty"`
	CompletedAt    *time.Time     `json:"completed_at,omitempty"`
	FailedAt       *time.Time     `json:"failed_at,omitempty"`
	DurationMs     int64          `json:"duration_ms"`
	RetryCount     int            `gorm:"default:0" json:"retry_count"`
}

// AutopilotWebhook represents a webhook endpoint for triggering autopilot tasks
type AutopilotWebhook struct {
	BaseModel
	WorkspaceID    uint64 `gorm:"not null;index" json:"workspace_id"`
	TaskID         uint64 `gorm:"not null;index" json:"task_id"`
	Endpoint       string `gorm:"size:255;not null;unique" json:"endpoint"`
	Secret         string `gorm:"size:100" json:"secret"`
	Status         string `gorm:"size:20;not null;default:active" json:"status"`
	AllowedIPs     json.RawMessage `gorm:"type:jsonb;default:'[]'" json:"allowed_ips"`
	MaxRequests    int    `gorm:"default:100" json:"max_requests"`
	RequestCount   int    `gorm:"default:0" json:"request_count"`
	LastUsedAt     *time.Time `json:"last_used_at,omitempty"`
}
